qiita_pet/uimodules/raw_data_tab.py

```diff
@@ -212,14 +175,26 @@ def render(self, prep, study_id, is_editable, ena_terms,
         preprocessing_status = prep.preprocessing_status
 
         if raw_data.filetype in ('SFF', 'FASTA'):
-            preprocess_form = Preprocess454ParametersForm()
+            all_params = Preprocessed454Params.get_all_parameters()
         elif raw_data.filetype == 'FASTQ':
-            preprocess_form = PreprocessIlluminaParametersForm()
+            all_params = PreprocessIlluminaParametersForm.get_all_parameters()
         else:
             raise ValueError("Don't know what to do but this exception will "
                              "never actually get shown anywhere because why "
                              "would you want to see tracebacks?")
 
+        preprocess_options = []
+        for param in all_params:
+            text = []
+            for k, v in viewitems(param):
+                if k in ('preprocessed_params_id', 'param_set_name'):
```

Comment:
God dammit :L

On (Feb-17-15|11:19), Daniel McDonald wrote:
> >          else:
> >              raise ValueError("Don't know what to do but this exception will "
> >                               "never actually get shown anywhere because why "
> >                               "would you want to see tracebacks?")
> >  
> > +        preprocess_options = []
> > +        for param in all_params:
> > +            text = []
> > +            for k, v in viewitems(param):
> > +                if k in ('preprocessed_params_id', 'param_set_name'):
> 
> For that size, a `set` would more likely increase overhead
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/859/files#r24844551